[Bug]: market-sim fuzz testing panics when amending order in place

### Problem encountered

Overnight market-sim fuzz tests panic when amending order in place.

Artifacts here - https://jenkins.vega.rocks/job/common/job/vega-market-sim-reinforcement/349/artifact/network_home/vega-sim-pzg_o7io/

```
panic: Failure after amend order from matching engine (amend-in-place)

goroutine 1748527 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x3?, 0x3?, {0x0?, 0x0?, 0xc021a9e0c0?})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00aa36820, {0xc0445fe9c0, 0x3, 0x3})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:262 +0x3ec
go.uber.org/zap.(*Logger).Panic(0xc050fcdd80?, {0x3d7a9aa?, 0xc050995c00?}, {0xc0445fe9c0, 0x3, 0x3})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/logger.go:258 +0x51
code.vegaprotocol.io/vega/core/execution/future.(*Market).orderAmendInPlace(0xc000a56600, 0xc02b947860, 0xc0508c6d00)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:3975 +0x5aa
code.vegaprotocol.io/vega/core/execution/future.(*Market).amendOrder(0xc000a56600, {0x4460f38, 0xc038a02000}, 0xc01c6af980, {0xc066f37bc0, 0x40})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:3770 +0x1898
code.vegaprotocol.io/vega/core/execution/future.(*Market).AmendOrderWithIDGenerator(0xc000a56600, {0x4460f38?, 0xc038a02000}, 0xc01c6af980, {0xc066f37bc0, 0x40}, {0x4431240?, 0xc00e9028a0?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:3456 +0x1bf
code.vegaprotocol.io/vega/core/execution.(*Engine).AmendOrder(0xc00015b080, {0x4460f38, 0xc038a02000}, 0xc01c6af980, {0xc066f37bc0, 0x40}, {0x4431240, 0xc00e9028a0})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/engine.go:1110 +0x3a4
code.vegaprotocol.io/vega/core/processor.(*App).DeliverAmendOrder(0xc000395c00, {0x4460f38, 0xc038a02000}, {0x448d7c0, 0xc00fe98c40}, {0xc048fde4c0, 0x40})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/processor/abci.go:1770 +0x29b
code.vegaprotocol.io/vega/core/processor.NewApp.addDeterministicID.func25({0x4460f38, 0xc038a02000}, {0x448d7c0, 0xc00fe98c40})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/processor/abci.go:542 +0x5e
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).SendTransactionResult.func26({0x4460f38, 0xc038a02000}, {0x448d7c0?, 0xc00fe98c40?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/processor/abci.go:605 +0x5d
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0xc00022c780, {0xb8?, 0x3934fe0?}, 0xc04dc6a000)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/abci/abci.go:173 +0x622
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0xc0100b3308?, {0x4460f00?, 0x68a8b20?}, 0xc00cc51e68?)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x25
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0xc000033680)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:198 +0x4a4
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0xc000033680, 0x77359400)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:275 +0x1ce
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:132 +0x25
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 1748525
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:131 +0x22f
```

### Observed behaviour

Panic

### Expected behaviour

No panic

### Steps to reproduce

```bash
replay
```


### Software version

527f78595a41404d5f62889e16958b24554da8b3

### Failing test

vega-market-sim fuzz test

### Jenkins run

https://jenkins.vega.rocks/job/common/job/vega-market-sim-reinforcement/349/

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "0000000000000000000000000000000000000000000000000000000000000000": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "blockchains.ethereumRpcAndEvmCompatDataSourcesConfig": "{\"configs\": []}",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "governance.proposal.transfer.maxClose": "8760h0m0s",
      "governance.proposal.transfer.maxEnact": "8760h0m0s",
      "governance.proposal.transfer.minClose": "1s",
      "governance.proposal.transfer.minEnact": "1s",
      "governance.proposal.transfer.minProposerBalance": "1",
      "governance.proposal.transfer.minVoterBalance": "1",
      "governance.proposal.transfer.requiredMajority": "0.5",
      "governance.proposal.transfer.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "limits.markets.proposePerpetualEnabled": "1",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0005",
      "market.fee.factors.makerFee": "0.0002",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "0000000000000000000000000000000000000000000000000000000000000000",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "1",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "0",
      "transfer.fee.maxQuantumAmount": "100",
      "transfer.feeDiscountMinimumTrackedAmount": "0.001",
      "transfer.feeDiscountDecayFraction": "0.5",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: Failure after amend order from matching engine (amend-in-place)

goroutine 1748527 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x3?, 0x3?, {0x0?, 0x0?, 0xc021a9e0c0?})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00aa36820, {0xc0445fe9c0, 0x3, 0x3})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/zapcore/entry.go:262 +0x3ec
go.uber.org/zap.(*Logger).Panic(0xc050fcdd80?, {0x3d7a9aa?, 0xc050995c00?}, {0xc0445fe9c0, 0x3, 0x3})
	/jenkins/GOPATH/pkg/mod/go.uber.org/zap@v1.25.0/logger.go:258 +0x51
code.vegaprotocol.io/vega/core/execution/future.(*Market).orderAmendInPlace(0xc000a56600, 0xc02b947860, 0xc0508c6d00)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:3975 +0x5aa
code.vegaprotocol.io/vega/core/execution/future.(*Market).amendOrder(0xc000a56600, {0x4460f38, 0xc038a02000}, 0xc01c6af980, {0xc066f37bc0, 0x40})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:3770 +0x1898
code.vegaprotocol.io/vega/core/execution/future.(*Market).AmendOrderWithIDGenerator(0xc000a56600, {0x4460f38?, 0xc038a02000}, 0xc01c6af980, {0xc066f37bc0, 0x40}, {0x4431240?, 0xc00e9028a0?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:3456 +0x1bf
code.vegaprotocol.io/vega/core/execution.(*Engine).AmendOrder(0xc00015b080, {0x4460f38, 0xc038a02000}, 0xc01c6af980, {0xc066f37bc0, 0x40}, {0x4431240, 0xc00e9028a0})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/engine.go:1110 +0x3a4
code.vegaprotocol.io/vega/core/processor.(*App).DeliverAmendOrder(0xc000395c00, {0x4460f38, 0xc038a02000}, {0x448d7c0, 0xc00fe98c40}, {0xc048fde4c0, 0x40})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/processor/abci.go:1770 +0x29b
code.vegaprotocol.io/vega/core/processor.NewApp.addDeterministicID.func25({0x4460f38, 0xc038a02000}, {0x448d7c0, 0xc00fe98c40})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/processor/abci.go:542 +0x5e
code.vegaprotocol.io/vega/core/processor.NewApp.(*App).SendTransactionResult.func26({0x4460f38, 0xc038a02000}, {0x448d7c0?, 0xc00fe98c40?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/processor/abci.go:605 +0x5d
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0xc00022c780, {0xb8?, 0x3934fe0?}, 0xc04dc6a000)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/abci/abci.go:173 +0x622
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0xc0100b3308?, {0x4460f00?, 0x68a8b20?}, 0xc00cc51e68?)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x25
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0xc000033680)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:198 +0x4a4
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0xc000033680, 0x77359400)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:275 +0x1ce
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:132 +0x25
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime in goroutine 1748525
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:131 +0x22f
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -35,6 +35,7 @@
 - [10792](https://github.com/vegaprotocol/vega/issues/10792) - Use correct market for market proposer bonus for spots.
 - [10795](https://github.com/vegaprotocol/vega/issues/10795) - Increase price monitoring triggers to 100 (missed validation).
 - [10800](https://github.com/vegaprotocol/vega/issues/10800) - Respect market list passed to transfer dispatch strategy when considering paid fees in applying reward cap. 
+- [10804](https://github.com/vegaprotocol/vega/issues/10804) - Handle amend in place revert on failure to update order margin such that we do not amend in place which may not be possible.
 
 ## 0.74.3
 

--- a/core/execution/future/market.go
+++ b/core/execution/future/market.go
@@ -3765,9 +3765,10 @@ func (m *Market) amendOrder(
 			pos, _ := m.position.GetPositionByPartyID(amendedOrder.Party)
 			if err := m.updateIsolatedMarginOnOrder(ctx, pos, amendedOrder); err == risk.ErrInsufficientFundsForMarginInGeneralAccount {
 				m.log.Error("party has insufficient margin to cover the order change, going to cancel all orders for the party")
-				_ = m.position.AmendOrder(ctx, amendedOrder, existingOrder)
-				// we're amending the order back in the order book so that when we unregister it, it would match what the position expects
-				m.orderAmendInPlace(amendedOrder, existingOrder)
+				// in this case we're now with the position with the amended order and the order book with the amended
+				// order which is consistent and will lead to the successful cancellation of all orders for the party.
+				// the reason we can't amend back here in place is that while decreasing size etc can be done in place
+				// increasing size (which is the inverse amend) cannot be done in place so we shouldn't attempt it.
 				return nil, nil, common.ErrMarginCheckFailed
 			}
 		}

